protoc

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+set -e
+
+if [ `uname -s` = "Darwin" ]; then
+  os=osx
+else
+  os=linux
+fi
+arch=`uname -m`
+
+protocbin=.protoc
+protocversion=3.0.0
+protocurl=https://github.com/google/protobuf/releases/download/v$protocversion/protoc-$protocversion-$os-$arch.zip
+
+if [ ! -f $protocbin ]; then
+  echo "downloading $protocbin" >&2
+  tmp=`mktemp -d`
```

Comment:
In my shell this requires a template argument -- e.g. `mktemp -d -t protoc.XXXXXX`

```
$ mktemp -d
usage: mktemp [-d] [-q] [-t prefix] [-u] template ...
       mktemp [-d] [-q] [-u] -t prefix 
$ echo $?
1
```